lib/auth0/mixins/httpproxy.rb

```diff
@@ -0,0 +1,57 @@
+module Auth0
+  module Mixins
+    # here's the proxy for Rest calls based on rest-client, we're building all request on that gem
+    # for now, if you want to feel free to use your own http client
+    module HTTPProxy
+      attr_accessor :headers, :base_uri, :timeout
+
+      # proxying requests from instance methods to HTTP class methods
+      %i(get post post_file put patch delete).each do |method|
+        define_method(method) do |path, body = {}|
+          safe_path = URI.escape(path)
+          body = body.delete_if { |_, v| v.nil? }
+          result = if [:get, :delete].include?(method)
+                      call(method, url(safe_path), timeout, add_headers({params: body}))
+                   elsif method == :post_file
+                      call(:post, url(safe_path), timeout, headers, body)
+                   else
+                      call(method, url(safe_path), timeout, headers, body.to_json)
```

Comment:
The difference between this line and line 16 is just the .to_json. What happen if we use the body without to_json here?